src/loaders/compute_tools/checkm2/checkm2.py

```diff
@@ -37,8 +46,28 @@ def _run_checkm2(
     run_command(command, log_dir if debug else None)
     end_time = time.time()
     print(f"Used {round((end_time - start) / 60, 2)} minutes to execute checkM2 predict "
-        + f"for {size} genomes"
-    )
+          + f"for {size} genomes"
+          )
+
+    _process_checkm2_result(output_dir, meta)
+
+
+def _process_checkm2_result(
+        output_dir: Path,
+        meta: Dict[str, str],
+):
+    # create metadata file before parsing the result
+    _create_metadata_file(meta, output_dir)
```

Comment:
The reason the metadata file was written after the tool execution finished was so that if tool execution failed partway it'd be clear to the parser that only partial results were available and it should fail. Since the jsonl files will now serve that purpose the meta files could be written out prior to the tool execution and you wouldn't have to pass the meta info into the tools, I think.

Reply:
Alternatively, if the meta file is only needed for this step, it looks like from the parsing code all you need is the tool identifier -> data-id mapping, so maybe you could just pass that around rather than creating a file.

Reply:
I have to write the metadata after tool execution because checkm2 needs a clean directory to execute. The metadata file is needed for other things like compute failure ids  as well.

Reply:
I see, ok.

In that case, after reading through things a bit more, it looks like the reason the checkm2 code needs the metadata file is just to get the tool identifier -> data id mapping. However, that mapping is already available in the `ids_to_files` argument. So what I'd do here is just pass the mapping to `read_genome_attri_result` (rather than parsing the metadata file in the method) and leave the metadata file generation where it was.

Reply:
what's wrong with keeping the existing logic and making minimum changes to code base? We need to generate the metadata file down the road anyways.

Reply:
What I'm talking about would reduce changes to the code base AFAICT, and it would mean someone implementing a new tool doesn't have to worry about writing the metadata file. 

Maybe we should huddle quickly on this one if we're still in disagreement

Reply:
👍 updated to use `ids_to_files`